### ISSUE TITLE
Wget2 regression: can't download from QT's develop sources

### ISSUE THREAD

**Issue by HinTak** [rockdaboot/wget2#328]
This can be reproduced locally, and also via github's CI (with github's fedora 40 container)

Somehow wget2 get confused by this url:
https://download.qt.io/development_releases/qt/6.8/6.8.0-beta1/src/submodules/qtwebengine-everywhere-src-6.8.0-beta1.tar.xz

And download is broken.

Wget 1 works fine.

**Comment by HinTak** [comment]
The CI problem is seen with https://github.com/HinTak/Qt6WE-OT-SVG/commit/003c2c51d95c725ed11c7b307e8462b418f3719d

Then I replaced wget2 with curl
https://github.com/HinTak/Qt6WE-OT-SVG/commit/46d9de06b33d09c2cb35a9abd1d8e54058cc0c9f .

**Comment by rockdaboot** [comment]
The Metalink protocol is offered by the web server and Wget2 selects it for downloading (it allows automatic content verification and recovering/resuming). The downside is that it takes longer as each download is split into (many) parts.

Here it fails at around part 413 with
```
wget2: nghttp2_session.c:2665: session_prep_frame: Assertion `0' failed.
```

A workaround (until this is fixed) is using `--no-metalink`.

TBH, I thought Metalink is mostly dead. Seeing it being supported is great.

**Comment by rockdaboot** [comment]
This issue is fixed by ec27488feadd44b5e126592bc18ff2441f8cae5a.

The Metalink chunked download is still triggered, but will work in combination with HTTP/2 now.
It still takes longer than using `--no-metalink`, the upside is that you have an automatic content verification via checksumming the downloaded file(s).

What we want to have in the future is a fast download *plus* an automatic checksum verification.
